control_panel_api/aws.py

```diff
@@ -37,6 +37,35 @@ def delete_policy(policy_arn):
     aws_api_client("iam").delete_policy(PolicyArn=policy_arn)
 
 
+def create_role(role_name, assume_role_policy):
+    """Creates IAM role with the given name"""
+
+    aws_api_client("iam").create_role(
+        RoleName=role_name,
+        AssumeRolePolicyDocument=json.dumps(assume_role_policy)
+    )
+
+
+def delete_role(role_name):
+    """Delete the given IAM role."""
+
+    _detach_role_policies(role_name)
+    aws_api_client("iam").delete_role(RoleName=role_name)
+
+
+def _detach_role_policies(role_name):
+    """Detaches all the policies from the given role"""
+
+    client = aws_api_client("iam")
```

Comment:
As per the other functions in this file you can chain the call instead of assigning to `client`. For consistency.

Reply:
This was coming from a time where inside the loop we also needed the client. 
Changing it now 👍

Reply:
Done: https://github.com/ministryofjustice/analytics-platform-control-panel/pull/19/commits/890936b97c71b5c9eb7f370e370cf651fc60617c